save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt

```diff
@@ -48,38 +48,50 @@ class FixPlugin(
 
         val fixPluginConfig = testConfig.pluginConfigs.filterIsInstance<FixPluginConfig>().single()
         val generalConfig = testConfig.pluginConfigs.filterIsInstance<GeneralConfig>().singleOrNull()
-        return files
-            .map { it.first() to it.last() }
-            .map { (expected, test) ->
-                val testCopy = createTestFile(test)
-                val execCmd = "${(generalConfig!!.execCmd)} ${fixPluginConfig.execFlags} $testCopy"
-                val executionResult = try {
-                    pb.exec(execCmd, null)
-                } catch (ex: ProcessExecutionException) {
-                    return@map TestResult(
-                        listOf(expected, test),
-                        Fail("${ex::class.simpleName}: ${ex.message}", ex::class.simpleName ?: "Unknown exception"),
-                        DebugInfo(null, ex.message, null)
-                    )
-                }
-                val stdout = executionResult.stdout.joinToString("\n")
-                val stderr = executionResult.stderr.joinToString("\n")
 
+        return files.chunked(fixPluginConfig.batchSize ?: 1).map { chunk ->
+            val pathMap = chunk.map { it.first() to it.last() }
+            val pathCopyMap = pathMap.map { (expected, test) -> expected to createTestFile(test) }
+            val testCopyNames = pathCopyMap.joinToString { (_, testCopy) -> testCopy.toString() }
+
+            val execCmd = "${(generalConfig!!.execCmd)} ${fixPluginConfig.execFlags} $testCopyNames"
+            val executionResult = try {
+                pb.exec(execCmd, null)
+            } catch (ex: ProcessExecutionException) {
+                return@map listOf(TestResult(
+                    pathMap.map { (expected, test) -> listOf(expected, test) }.flatten(),
```

Comment:
```suggestion
                return@map chunk.map {
                    TestResult(it,
```
I think an additional `pathMap` collection is redundant, because `chunk` already is a list of groups of files,